### PR TITLE
Stuff1

### PR DESCRIPTION
**Pull request by SimonCropp**
mostly whitespace to reduce noise in future PRs

### CODE REVIEW

**Comment by alistairjevans** [comment]
Thanks @SimonCropp!

I was unsure whether to enable nullable constraints globally, due to the extra markers required in test projects, but perhaps having to explicitly send null is better? 

As for the whitespace, there are a couple of places where end-of-line whitespace remains important, mostly inside literal code blocks used for language tests, because the test language needs to be able to tokenise whitespace, so I may have to leave that whitespace behind.

**Comment by SimonCropp** [comment]
i left some whitespace in tests. did i miss some?

File: tests/AutoStep.Tests/Language/Test/Parsing/TableTests.cs [hunk 1]
```diff
@@ -198,9 +197,9 @@ public async Task UnterminatedTableError()
                     Scenario: My Scenario
 
                         Given this step has a table:
-                            | heading1         | 
+                            | heading1         |
                             | 123              |
-                            | 123.50
```
Comment (alistairjevans) [inline tests/AutoStep.Tests/Language/Test/Parsing/TableTests.cs:203]: This whitespace should be kept (due to testing an unterminated row).

File: tests/AutoStep.Tests/Language/Test/Parsing/TableTests.cs [hunk 2]
```diff
@@ -261,10 +259,10 @@ public async Task MultipleUnterminatedRows()
...
```
Comment (alistairjevans) [inline tests/AutoStep.Tests/Language/Test/Parsing/TableTests.cs:266]: This whitespace should be kept (due to testing an unterminated row).

File: tests/AutoStep.Tests/Language/Test/Parsing/TableTests.cs [hunk 3]
```diff
@@ -322,9 +318,9 @@ public async Task TableBlankRowGap()
                     Scenario: My Scenario
 
                         Given this step has a table:
-                            | heading1         |
```
Comment (alistairjevans) [inline tests/AutoStep.Tests/Language/Test/Parsing/TableTests.cs:325]: This whitespace should be kept (due to testing a blank row).

**Review by alistairjevans — CHANGES_REQUESTED** [review]
There's just a couple of whitespace issues in a few of the table tests.

Also, are you ok to fix the nullable reference issues that have appeared due to turning on nullable in the tests?

File: src/AutoStep/Language/Test/Visitors/FileVisitor.cs [hunk 4]
```diff
@@ -68,26 +68,26 @@ public override FileElement VisitFeatureBlock([NotNull] AutoStepParser.FeatureBl
             // Clear the set of feature scenario names.
             featureScenarioNames.Clear();
 
-            if (Result.Feature != null)
+            if (Result!.Feature != null)
```
Comment (alistairjevans) [inline src/AutoStep/Language/Test/Visitors/FileVisitor.cs:71]: Were you still getting nullability warnings in this file? The Debug.Assert should be taking care of that already....

Comment (SimonCropp) [reply src/AutoStep/Language/Test/Visitors/FileVisitor.cs:71]: not in release mode

Comment (alistairjevans) [reply src/AutoStep/Language/Test/Visitors/FileVisitor.cs:71]: Ah, of course. 🤦‍♂️

**Comment by SimonCropp** [comment]
seems i have fixed the warnings?

**Comment by SimonCropp** [comment]
this will need a squash on merge

**Review by alistairjevans — APPROVED** [review]

**Comment by alistairjevans** [comment]
Yes, all looks good to me now. 

Thanks for sticking with the nullable changes, I know they can be a pain.

You happy for me to merge this?

**Comment by SimonCropp** [comment]
yep ready to merge
